modules/twinklewarn.js

```diff
@@ -1202,31 +1203,34 @@ Twinkle.warn.callback.change_category = function twinklewarnCallbackChangeCatego
 	// Trigger custom label/change on main category change
 	Twinkle.warn.callback.change_subcategory(e);
 
-	// Build select menu with jquery.chosen
-	if (!Twinkle.getPref('oldSelect')) {
+	// Use select2 to make the select menu searchable
+	if (!Twinkle.getPref('oldSelect') && $.fn.select2) {
 		$('select[name=sub_group]')
-			.chosen({width: '100%', search_contains: true})
+			.select2({ width: '100%' })
 			.change(Twinkle.warn.callback.change_subcategory);
 
+		// if main_group is level1/2/3/4/4im, the ul has option groups,
+		if ($('select[name=main_group]').val().indexOf('level') === 0) {
+			Morebits.select2SearchHighlights($('select[name=sub_group]'), true);
+		} else {
+			Morebits.select2SearchHighlights($('select[name=sub_group]'), false);
+		}
+
 		mw.util.addCSS(
-			// Force chosen select menu to display over the dialog while overflowing
-			// based on https://github.com/harvesthq/chosen/issues/1390#issuecomment-21397245
-			'.ui-dialog.morebits-dialog .morebits-dialog-content { overflow:visible !important; }' +
-			'.ui-dialog.morebits-dialog { overflow: inherit !important; }' +
+			// prevent dropdown from appearing behind the dialog
+			'.select2-container { z-index: 10000; }' +
 
-			// Increase height to match that of native select
-			'.morebits-dialog .chosen-drop .chosen-results { max-height: 300px; }' +
-			'.morebits-dialog .chosen-drop { max-height: 338px; }' +
+			// Increase height
+			'.select2-container .select2-dropdown .select2-results > .select2-results__options { max-height: 350px; }' +
 
-			// Remove padding
-			'.morebits-dialog .chosen-drop .chosen-results li { padding-top: 0px; padding-bottom: 0px; }'
-		);
-		// (Hopefully) temporary fixes for visual issues caused by the above CSS, see #665 and #667
-		$('.ui-resizable-handle').remove();
-		// Jump to extend the dialog to include any previewing content
-		// without scrolling and prevent resizing
-		mw.util.addCSS('.ui-dialog.morebits-dialog .morebits-dialog-content { max-height: max-content !important; }');
+			// Reduce padding
+			'.select2-results .select2-results__option { padding-top: 1px; padding-bottom: 1px; }' +
+			'.select2-results .select2-results__group { padding-top: 1px; padding-bottom: 1px; } ' +
 
+			// Reduce font size
```

Comment:
```suggestion
			// Increase font size
```

Reply:
This must be platform-dependent, because I was seeing a default font size of a massive 16px. Let's change the word to "Adjust".

Reply:
What percent was that?  I can't check  my system tonight but it was around 11.5px iirc.